### DISCUSSION TITLE
[🐞] onSubmitCompleted$ is always triggered, even when fails

### DISCUSSION THREAD

**Discussion by keuller** [QwikDev/qwik-evolution#301] [Proposals For Qwik]
### Which component is affected?

Qwik Runtime

### Describe the bug

According to the documentation `onSubmitCompleted$` should be executed right after the action is executed successfully and returns some data. The issue: it is always executed no matter Zod validation fails or I return `fail` inside `actionRoute$`.

### Reproduction

N/A

### Steps to reproduce

You can use this code:
```jsx
import { component$ } from '@builder.io/qwik';
import { Form, routeAction$, zod$, z } from '@builder.io/qwik-city';

export const useSaveAction = routeAction$(async (lead, { request, fail }) => {
    console.log('lead:', lead);
    
    return {
    }
}, zod$({
    name: z.string().min(3),
    email: z.string().email(),
}));

export default component$(() => {
   const action = useSaveAction();

   return (
   <Form id='leadForm' action={action}
           onSubmitCompleted$={(ev) => console.log('onSubmitCompleted', ev.detail)}>

      <label for="id" class="text-sm text-slate-500">{ props.label }</label>
       <input type="text" name="name"
            class="border border-slate-500 rounded-md p-2" 
            maxLength={50} />

       <input type="email" name="email"
            class="border border-slate-500 rounded-md p-2" 
            maxLength={100} />

      <div class="col-span-2 py-2">
          <button type="submit" name="_intent" 
              value="create"
              disabled={action.isRunning}
              class="outline-none bg-indigo-500 hover:bg-indigo-600 px-6 py-2 rounded-md text-sm text-white disabled:bg-slate-400">
              <span>{action.isRunning ? 'Saving...' : 'Save'}</span>
          </button>
       </div>
   </Form>)
});

### System Info

```shell
System: Mac OSX Ventura 13.3
Qwik version: 1.1.4
Qwik-City: 1.1.4
Typescript: 5.0.4
Node: 16.15.0
```

**Comment by keuller** [comment]
I opened the PR [4397](https://github.com/BuilderIO/qwik/pull/4397) that fix it. 
cc @manucorporat

**Comment by manucorporat** [comment]
Wondering if this is an bug.
Might be important to also react when the submit failed.


We could add add `onSubmitSucceed`, `onSubmitFailed`

**Comment by keuller** [comment]
HI @manucorporat. 

What is exactly your doubt if it is a bug or not? Can you explain in detail your thoughts?

Regarding the submit failure, I ran it many times the tests locally and it works, tbh I don't have any clue why it is failing via CI. I checked out the logs but I didn't find out the root cause.

I will review the code again and try to apply your suggestion: add `onSubmitSucceed` or `onSubmitFailed`.

Thank you to take a look at it.

**Comment by PatrickJS** [comment]
the API should be reworked maybe in v2
some fixes done in https://github.com/QwikDev/qwik/pull/6241

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.
